ddsrecorder/src/cpp/tool/DdsRecorder.cpp

```diff
@@ -147,6 +155,24 @@ void DdsRecorder::trigger_event()
     mcap_handler_->trigger_event();
 }
 
+participants::McapHandlerStateCode DdsRecorder::recorder_to_handler_state_(
+        const DdsRecorderStateCode& recorder_state)
+{
+    switch (recorder_state)
+    {
+        case DdsRecorderStateCode::RUNNING:
+            return participants::McapHandlerStateCode::RUNNING;
+
+        case DdsRecorderStateCode::PAUSED:
+            return participants::McapHandlerStateCode::PAUSED;
+
+        case DdsRecorderStateCode::STOPPED:
+        case DdsRecorderStateCode::SUSPENDED:
+        default:  // TODO: maybe better throw exception????????????????????????????
```

Comment:
I agree with you. Throw an exception (always captured) an do not change the recorder state.